lib/SILOptimizer/Transforms/ArrayElementValuePropagation.cpp

```diff
@@ -207,8 +232,12 @@ bool ArrayAllocation::recursivelyCollectUses(ValueBase *Def) {
     // Check array semantic calls.
     ArraySemanticsCall ArrayOp(User);
     if (ArrayOp && ArrayOp.doesNotChangeArray()) {
+      if (ArrayOp.getKind() == ArrayCallKind::kAppendContentsOf)
```

Comment:
kAppendContentsOf _does_ change the self-array. But you care about it's source-argument (and not about the self argument).
You can do something like

```
if (ArrayOp) {
  if (ArrayOp.getKind() == ArrayCallKind::kAppendContentsOf) {
    // ArrayOp is the "newElements" operand
    ...
    continue;
  } else if (ArrayOp.getKind() == ArrayCallKind::kGetElement) {
   ...
    continue;
  } else if (ArrayOp.doesNotChangeArray()) {
    continue;
  }
}
```